### ISSUE TITLE
Removing a zone from config deletes pre-existing records

### ISSUE THREAD

**Issue by hundt** [akamai/terraform-provider-akamai#28]
When adding or removing records, pre-existing records are left alone. But when removing the entire zone from a config, most pre-existing records are deleted (everything but SOA I think).

To me this seems like a bug, because I think deleting a resource should be the inverse of adding a resource. When it is not, you get situations like the following:

1. example.com. zone has pre-existing records (say, a TXT) and is not referenced in any terraform configs
2. `example.tf` is created with the contents like the in the [README](https://github.com/akamai/terraform-provider-akamai#akamai_fastdns_zone). 
3. `terraform apply`: new CNAME and A records added and TXT record left alone.
4. zone removed from `example.tf`
5. `terraform apply`: the CNAME, A, and TXT records are all deleted.

This seems confusing to users, and also limits use of the plugin to situations where the entire zone except for the SOA is defined in the terraform file.

One alternative is to treat a "delete" like an update to an empty zone and use the existing merge logic. I implemented that here: https://github.com/hundt/terraform-provider-akamai/commit/dedf1067a6841149c1d1a9dd117169c769877836

If you like that approach I can turn it into a pull request.

**Comment by dshafik** [comment]
I agree with you that this is the correct approach (definitely the "least surprise"). Please do create a PR and we can review and get it merged!

Thanks for your contribution!
